### PR TITLE
Initial HTTPS support

### PR DESCRIPTION
**Pull request by sameo**
This is the initial support for HTTPS:

* rusttls (i.e. ring, i.e. boringssl) based.
* HTTPS is enabled by default
* `--insecure-http` disables HTTPS and needs to be explicitly set

### CODE REVIEW

File: src/api_server/src/lib.rs [hunk 1]
```diff
@@ -71,20 +79,73 @@ impl ApiServer {
...
+        private_key: Option<PathBuf>,
+        certificate: Option<PathBuf>,
+        insecure: bool,
         attestation_service: Arc<AttestationService>,
         http_timeout: i64,
-    ) -> Self {
-        ApiServer {
+    ) -> Result<Self> {
+        if !insecure && (private_key.is_none() || certificate.is_none()) {
+            return Err(anyhow!("Missing HTTPS credentials"));
```
Comment (Xynnn007) [inline src/api_server/src/lib.rs:89]: NIT: we can use `anyhow::bail!("Missing HTTPS credentials")` to make shorter

File: src/api_server/src/lib.rs [hunk 2]
```diff
@@ -71,20 +79,73 @@ impl ApiServer {
...
+            File::open(
+                self.private_key
+                    .clone()
+                    .ok_or(anyhow!("Missing private key"))?,
+            )
+            .map_err(anyhow::Error::from)?,
+        );
+
+        let cert_chain = certs(cert_file)
+            .map_err(anyhow::Error::from)?
```
Comment (Xynnn007) [inline src/api_server/src/lib.rs:123]: Same as mentioned. Could we directly use `?` to replace `.map_err(anyhow::Error::from)?` ?

**Review by Xynnn007 — COMMENTED** [review]
Thanks a lot for this! Only some nits

File: src/api_server/src/lib.rs (hunk 1 above)
Comment (sameo) [reply src/api_server/src/lib.rs]: Thanks, fixed now.

File: src/api_server/src/lib.rs [hunk 3]
```diff
@@ -71,20 +79,73 @@ impl ApiServer {
...
+    }
+
+    fn tls_config(&self) -> Result<ServerConfig> {
+        let cert_file = &mut BufReader::new(
+            File::open(
+                self.certificate
+                    .clone()
+                    .ok_or(anyhow!("Missing certificate"))?,
+            )
+            .map_err(anyhow::Error::from)?,
```
[the remark this answers is not in the document]
Comment (sameo) [reply src/api_server/src/lib.rs]: you're right, fixed.

[the remark this answers is not in the document]
Comment (sameo) [reply src/api_server/src/lib.rs]: Right as well :)

File: Cargo.toml [hunk 4]
```diff
@@ -12,7 +12,7 @@ edition = "2021"
 api-server = { path = "src/api_server" }
 anyhow = "1.0"
 async-trait = "0.1.31"
-attestation-service = { git = "https://github.com/confidential-containers/attestation-service.git" }
+attestation-service = { git = "https://github.com/confidential-containers/attestation-service.git" , default-features = false, features =[ "rvps-server", "in-toto" ] }
```
[the remark this answers is not in the document]
Comment (sameo) [reply Cargo.toml:15]: Ok, I did not realize that. Thanks for the heads up, I guess it's fine if I keep it like that or do you prefer that I switch to `rvps-proxy`?

[the remark this answers is not in the document]
Comment (Xynnn007) [reply Cargo.toml:15]: Sorry my fault. It is not clear enough.  It is totally fine to keep it now. I'll open another PR to switch to `rvps-proxy`. : )

**Review by Xynnn007 — APPROVED** [review]
Thanks for adopting all the comments. This PR looks great to me! : )

**Review by jialez0 — APPROVED** [review]
LGTM. Thanks! @sameo 

At present, we have not determined a scheme to transfer KBS certificates to CC-KBC, but we will discuss and implement that in the near future.

**Comment by bpradipt** [comment]
Any plans to update the doc with an example execution command, say using a self-signed certificate ?
Please don't consider this as a blocker to merge this PR.

File: Cargo.toml (hunk 4 above)
Comment (jialez0) [inline Cargo.toml:15]: Maybe we should also enable `"all-verifier"` feature of attestation-service here?

Comment (sameo) [reply Cargo.toml]: I changed that part now. The default cargo build and make kbs are unmodified, and added a new `kbs-no-verifier` makefile target.

**Comment by sameo** [comment]
> Any plans to update the doc with an example execution command, say using a self-signed certificate ? Please don't consider this as a blocker to merge this PR.

Good point. I'll open an issue to document self-signed cert generation once this PR is merged.

**Comment by Xynnn007** [comment]
FYI The build error is related to this issue https://github.com/confidential-containers/attestation-service/issues/62

**Comment by jialez0** [comment]
@sameo Now https://github.com/confidential-containers/attestation-service/pull/63 is merged, can you run CI again and can we try to get this PR merged?

**Review by jialez0 — APPROVED** [review]

**Review by jialez0 — APPROVED** [review]

**Review by jialez0 — APPROVED** [review]

**Review by Xynnn007 — APPROVED** [review]

**Pull request merged**